raiden/smart_contracts/TokenNetwork.sol

```diff
@@ -0,0 +1,568 @@
+        require(closing_requests[channel_identifier].settle_block_number > 0);
+        _;
+    }
+
+    modifier stillTimeout(uint channel_identifier) {
+        require(closing_requests[channel_identifier].settle_block_number > block.number);
+        _;
+    }
+
+    modifier timeoutOver(uint channel_identifier) {
+        require(closing_requests[channel_identifier].settle_block_number <= block.number);
+        _;
+    }
+
+    // Note: we use the settleTimeout to check if the channel is open. It must always be > 0.
+    modifier settleTimeoutValid(uint timeout) {
+        require(timeout >= 6 && timeout <= 2700000);
+        _;
+    }
+
+    /*
+     *  Constructor
+     */
+
+    function TokenNetwork(address _token_address) public {
+        require(_token_address != 0x0);
+        require(contractExists(_token_address));
+
+        token = Token(_token_address);
+
+        // Check if the contract is indeed a token contract
+        require(token.totalSupply() > 0);
+    }
+
+    /*
+     *  Public functions
+     */
+
+    /// @notice Opens a new channel between `participant1` and `participant2`.
+    /// Can be called by anyone.
+    function openChannel(
+        address participant1,
+        address participant2,
+        uint settle_timeout)
+        settleTimeoutValid(settle_timeout)
+        public
+        returns (uint)
+    {
+        require(participant1 != 0x0);
+        require(participant2 != 0x0);
+        require(participant1 != participant2);
+
+        // Increase channel index counter
+        last_channel_index += 1;
+
+        require(channels[last_channel_index].settle_timeout == 0);
+        require(!channels[last_channel_index].participants[participant1].initialized);
+        require(!channels[last_channel_index].participants[participant2].initialized);
+
+        // Store channel information
+        channels[last_channel_index] = Channel({settle_timeout: settle_timeout});
+
+        // Mark the channel participants
+        // We use this in deposit to ensure the beneficiary is a channel participant
+        channels[last_channel_index].participants[participant1].initialized = true;
+        channels[last_channel_index].participants[participant2].initialized = true;
+
+        ChannelOpened(last_channel_index, participant1, participant2, settle_timeout);
+
+        return last_channel_index;
+    }
+
+    /// @notice Deposit tokens to an already existent channel.
+    /// Can be called by anyone.
+    function deposit(
+        uint channel_identifier,
+        address beneficiary,
+        uint256 added_amount)
+        public
+        returns (bool)
+    {
+        Channel storage channel = channels[channel_identifier];
+
+        // Channel must be open and beneficiary must be one of the participants
+        require(channel.participants[beneficiary].initialized);
+
+        // Channel cannot be closed
+        require(closing_requests[channel_identifier].settle_block_number == 0);
```

Comment:
To be honest, at first I was thinking that it would save gas cost because you only initialize that when you close a channel. Under this assumption we use this in uRaiden.  
But I realized that I have to make some gas tests on `Struct`s. I thought that because they have the types defined, they occupy all storage slots needed from the beginning. I may be wrong and this might happen: if you have a defined variable in the structure but don't initialize it (`closing_address` ..) you don't pay gas.

Reply:
Yeah, I'm not 100% sure either